src/api/CortexClient.ts

```diff
@@ -91,17 +91,27 @@ export class CortexClient implements CortexApi {
       ? entities.map(entity => applyCustomMappings(entity, customMappings))
       : entities;
 
-    if (gzipContents) {
-      return await this.postWithGzipBody(`/api/backstage/v1/entities/sync`, {
-        entities: withCustomMappings,
-        teamOverrides,
-      });
-    } else {
-      return await this.post(`/api/backstage/v1/entities/sync`, {
-        entities: withCustomMappings,
-        teamOverrides,
-      });
+    const post = async (path: string, body?: any) => {
+        return gzipContents ? await this.postVoidWithGzipBody(path, body) : await this.postVoid(path, body)
     }
+
+    await this.postVoid('/api/backstage/v2/entities/sync-init')
+    let maxListSize = Math.max(
+      withCustomMappings.length,
+      teamOverrides?.teams?.length ?? 0,
+      teamOverrides?.relationships?.length ?? 0
+    )
+    for (let i = 0; i < maxListSize; i+= CHUNK_SIZE) {
+      await post(`/api/backstage/v2/entities/sync-chunked`, {
+        entities: withCustomMappings.slice(i, i + CHUNK_SIZE),
```

Comment:
Should `withCustomMappings` and `teamOverrides` have the same length? Because we're chunking them the same.